src/transformStateWithClones.js

```diff
@@ -8,6 +8,38 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let clone = {};
+
+  for (let i = 0; i < actions.length; i++) {
+    if (result.length === 0) {
+      clone = Object.assign({}, state);
+    } else {
+      clone = Object.assign({}, result[result.length - 1]);
+    }
+
+    switch (actions[i].type) {
+      case 'addProperties':
+        result.push(Object.assign({}, clone, actions[i].extraData));
```

Comment:
You can assing directly into clone, as it can be modified as you wish, only original state must remain intact

```suggestion
        result.push(Object.assign(clone, action.extraData));
```